airflow/runtimes/docker.go

```diff
@@ -0,0 +1,83 @@
+package runtimes
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/briandowns/spinner"
+)
+
+const (
+	defaultTimeoutSeconds = 60
+	tickNum               = 500
+	open                  = "open"
+	timeoutErrMsg         = "timed out waiting for docker"
+	dockerOpenNotice      = "We couldn't start the docker engine automatically. Please start it manually and try again."
+)
+
+// DockerInitializer is a struct that contains the functions needed to initialize Docker.
+// The concrete implementation that we use is DefaultDockerInitializer below.
+// When running the tests, we substitute the default implementation with a mock implementation.
+type DockerInitializer struct {
```

Comment:
This looks like it should be an interface, what do you think?

Reply:
Ah yea good call, meant to do that. It's now an interface and I've updated the mock implementation and tests accordingly.